Index: linux-xlnx-xlnx_rebase_v6.6_LTS_2024.1/arch/arm/boot/dts/xilinx/Makefile
===================================================================
--- linux-xlnx-xlnx_rebase_v6.6_LTS_2024.1.orig/arch/arm/boot/dts/xilinx/Makefile
+++ linux-xlnx-xlnx_rebase_v6.6_LTS_2024.1/arch/arm/boot/dts/xilinx/Makefile
@@ -1,5 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0
 dtb-$(CONFIG_ARCH_ZYNQ) += \
+	zynq-arty-z7.dtb \
 	zynq-cc108.dtb \
 	zynq-ebaz4205.dtb \
 	zynq-microzed.dtb \
Index: linux-xlnx-xlnx_rebase_v6.6_LTS_2024.1/arch/arm/boot/dts/xilinx/zynq-arty-z7.dts
===================================================================
--- /dev/null
+++ linux-xlnx-xlnx_rebase_v6.6_LTS_2024.1/arch/arm/boot/dts/xilinx/zynq-arty-z7.dts
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: GPL-2.0+
+/dts-v1/;
+#include "zynq-7000.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "Digilent Arty Z7 board";
+	compatible = "digilent,zynq-arty-z7", "xlnx,zynq-7000";
+
+	aliases {
+		ethernet0 = &gem0;
+		serial0 = &uart0;
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x40000000>;
+	};
+
+	chosen {
+		bootargs = "";
+		stdout-path = "serial0:115200n8";
+	};
+
+	/*gpio-leds {
+		compatible = "gpio-leds";
+
+		led-ld4 {
+			label = "zynq-zybo-z7:green:ld4";
+			gpios = <&gpio0 7 GPIO_ACTIVE_HIGH>;
+		};
+	};*/
+
+	usb_phy0: phy0 {
+		#phy-cells = <0>;
+		compatible = "ulpi-phy";
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
+		//reset-gpios = <&gpio0 46 GPIO_ACTIVE_LOW>;
+	};
+};
+
+&clkc {
+	ps-clk-frequency = <50000000>;
+};
+
+&gem0 {
+	status = "okay";
+	phy-mode = "rgmii-id";
+	phy-handle = <&ethernet_phy>;
+
+	ethernet_phy: ethernet-phy@0 {
+		reg = <0>;
+		device_type = "ethernet-phy";
+	};
+};
+
+&sdhci0 {
+	status = "okay";
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&usb0 {
+	status = "okay";
+	dr_mode = "host";
+	usb-phy = <&usb_phy0>;
+};
